lib/rbs/test/guaranteed.rb

```diff
@@ -0,0 +1,36 @@
+module RBS
+  module Test
+    module Guaranteed
+      IS_AP = ::Kernel.instance_method(:is_a?)
+      private_constant :IS_AP
+
+      def self.value_is_a?(value, kind)
+        IS_AP.bind_call(value, kind)
+      end
+
+      module Inspect
+        EQUAL = ::BasicObject.instance_method(:equal?)
+        INSPECT = ::Kernel.instance_method(:inspect)
+        private_constant :EQUAL, :INSPECT
+
+        module_function def guaranteed_inspect(obj)
+          obj.inspect
+        rescue NoMethodError => err
+          raise unless err.name == :inspect && EQUAL.bind_call(obj, err.receiver)
+          GUARANTEED_INSPECT.bind_call(obj)
+        end
+
+        def inspect
```

Comment:
Custom inspect; We could update it to mamke it prettier, but I think this is good enough for now.

Reply:
```ruby
irb(main):001> BasicObject.new
=> #<BasicObject:0x00007f63cfad6260>
irb(main):002> _.inspect
(irb):2:in `<main>': undefined method `inspect' for #<BasicObject:0x00007f63cfad6260> (NoMethodError)
        from /usr/local/lib/ruby/gems/3.2.0/gems/irb-1.8.1/exe/irb:9:in `<top (required)>'
        from /usr/local/bin/irb:25:in `load'
        from /usr/local/bin/irb:25:in `<main>'
```
I wonder how does IRB do it

Can also try binding `Kernel#inspect` on it

P.S. that could be how IRB does it
```
irb(main):001> class B < BasicObject; attr_accessor :b end
=> [:b, :b=]
irb(main):002> b = B.new
=> #<B:0x00007f18d3b123d8>
irb(main):003> b.b = 1
=> 1
irb(main):004> b
=> #<B:0x00007f18d3b123d8 @b=1>
irb(main):005> b.b = b
An error occurred when inspecting the object: #<NoMethodError: undefined method `inspect' for #<B:0x00007f18d3b123d8>>
An error occurred when running Kernel#inspect: #<NoMethodError: undefined method `inspect' for #<B:0x00007f18d3b123d8>>
…
```